src/app/pages/things/details/things.details.component.ts

```diff
@@ -128,6 +128,19 @@ export class ThingsDetailsComponent implements OnInit, OnDestroy {
     });
   }
 
+  onChangeDate(event) {
+    const from = new Date(event.from).getTime() / 1000;
+    const to = new Date(event.to).getTime() / 1000;
+
+    if (from && to) {
+      this.messagesService.getMessages(this.connectedChans[0].id, this.thing.key, this.thing.id, '', 0 , 100, from, to).subscribe(
```

Comment:
Why do we use exactly `0` and `100`. What happens if there is more than `100` messages in that time range? Anyway, this should be consts. Besides, maybe we can treat `to` as optional and fetch from `from` till the end. That way you save the user of trouble of having always to select `to` (this is just a suggestion).

Reply:
Right, we should probably emit an event for pagination